app/src/androidTest/java/org/systers/mentorship/LoginActivityTest.kt

```diff
@@ -0,0 +1,160 @@
+     *  with empty username and password fields.
+     */
+    @Test
+    fun testLoginButtonClickedWhenUsernameAndPasswordAreEmpty() {
+
+        // Type empty string in both the editTexts
+        findEditTextInTextInputLayout(R.id.tiUsername).perform(typeText(""), closeSoftKeyboard())
+        findEditTextInTextInputLayout(R.id.tiPassword).perform(typeText(""), closeSoftKeyboard())
+
+        // Perform Click operation on Login Button
+        onView(withId(R.id.btnLogin)).perform(click())
+
+        // Checks that the error message in both the editTexts appears after button click
+        onView(withId(R.id.tiUsername)).check(matches(hasTextInputLayoutErrorText(EMPTY_USERNAME_ERROR)))
+        onView(withId(R.id.tiPassword)).check(matches(hasTextInputLayoutErrorText(EMPTY_PASSWORD_ERROR)))
+
+    }
+
+    /** This test checks that error messages are shown after the Login button is clicked
+     *  with empty username and a password.
+     */
+    @Test
+    fun testLoginButtonClickedWhenUsernameIsEmptyAndPasswordIsFilled() {
+        //Type in no username
+        findEditTextInTextInputLayout(R.id.tiUsername).perform(typeText(""), closeSoftKeyboard())
+
+        //Type in a password
+        findEditTextInTextInputLayout(R.id.tiPassword).perform(typeText(TEST_PASSWORD), closeSoftKeyboard())
+
+        // Perform Click operation on Login Button
+        onView(withId(R.id.btnLogin)).perform(click())
+
+        //Check for error message on username and not on password
+        onView(withId(R.id.tiUsername)).check(matches(hasTextInputLayoutErrorText(EMPTY_USERNAME_ERROR)))
+        onView(withId(R.id.tiPassword)).check(matches(not(hasTextInputLayoutErrorText(EMPTY_PASSWORD_ERROR))))
+    }
+
+    /** This test checks that error messages are shown after the Login button is clicked
+     *  with a username and an empty password.
+     */
+    @Test
+    fun testLoginButtonClickedWhenUsernameIsFilledAndPasswordIsEmpty() {
+        //Type in a username
+        findEditTextInTextInputLayout(R.id.tiUsername).perform(typeText(TEST_USERNAME), closeSoftKeyboard())
+
+        //Type in no password
+        findEditTextInTextInputLayout(R.id.tiPassword).perform(typeText(""), closeSoftKeyboard())
+
+        // Perform Click operation on Login Button
+        onView(withId(R.id.btnLogin)).perform(click())
+
+        //Check for no error message on username with an error message on password
+        onView(withId(R.id.tiUsername)).check(matches(not(hasTextInputLayoutErrorText(EMPTY_USERNAME_ERROR))))
+        onView(withId(R.id.tiPassword)).check(matches(hasTextInputLayoutErrorText(EMPTY_PASSWORD_ERROR)))
+
+    }
+
+    /** This test checks that a snackbar message appears when a user enters a bad username and password
+     *  with a username and an empty password.
+     */
+    @Test
+    fun testLoginButtonClickedWhenUsernameAndPasswordAreBad() {
+        //Type in a username
```

Comment:
```suggestion
        // Type in a username
```

Follow this example in other one liner comments please